### PR TITLE
[WFCORE-254] Removed the org.apache.commons.logging module as it will…

### PR DESCRIPTION
**Pull request by jamezp**
… be provided by WildFly Core.

The `org.apache.commons.logging` module has moved to WildFly Core in https://github.com/wildfly/wildfly-core/pull/1799. This uses a new project for the implementation. The `org.slf4j.jcl-over-slf4j` was left here for backwards compatibility. It's possible since it's a private API we should consider removing it though or at least having it delegate to the new `org.jboss.logmanager.commons.logging` module.

Once a new version of core is released I will integrate this with EAP 7.

### CODE REVIEW

**Comment by jamezp** [comment]
This CI job is running this branch with the core branch required https://ci.wildfly.org/viewLog.html?buildId=26612&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperiments

**Comment by bstansberry** [comment]
If we leave org.slf4j.jcl-over-slf4j we're basically saying leave it until EAP 8 as there's no compatibility reason to remove it in a later EAP 7 minor that isn't just as valid today. It's private and we're quite clear about the meaning of that, so we're talking about when we'd stop going above and beyond our stated policy.

**Comment by jamezp** [comment]
Okay. I don't mind deleting it at all. I'll go ahead and update both this to remove the module for this PR and the EAP PR.

File: client/jms/pom.xml [hunk 1]
```diff
@@ -130,6 +130,11 @@
             <groupId>org.slf4j</groupId>
             <artifactId>jcl-over-slf4j</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
```
Comment (jamezp) [inline client/jms/pom.xml]: I only added this as it seemed like the dependency should be explicit rather than transit from `org.slf4j:jcl-over-slf4j`.

**Comment by jamezp** [comment]
Updated to remove the `org.slf4j.jcl-over-slf4` module. The core upgrade has been done as well.

**Pull request merged**